### DISCUSSION TITLE
Can fpdf2 be used to print pdfs?

### DISCUSSION THREAD

**Discussion by mak448a** [py-pdf/fpdf2#1149] [Q&A] [answered]
I saw something like this on reddit for PyFPDF. https://www.reddit.com/r/learnpython/comments/16fnei4/comment/kjhn1lo/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Comment by andersonhc** [comment, marked as answer]
> I saw something like this on reddit for PyFPDF. https://www.reddit.com/r/learnpython/comments/16fnei4/comment/kjhn1lo/

`fpdf2` is a library you can use to generate PDF files.

If you need to produce a PDF and then print it, `fpdf2` might be part of your solution. If you already have PDF files and just need to print them, then you don't need `fpdf2`.

**Reply by mak448a** [reply]
So is it possible to print a pdf with fpdf2?

**Reply by joren-dev** [reply]
You can create a PDF using fpdf2, but you can't print it directly using fpdf2. Printing needs a different approach. You have to use another Python package or access your operating system directly to communicate with the printer driver installed on your computer. Then, you send the generated PDF to the printer.

**Reply by mak448a** [reply]
Thanks. I just used `lpr` on my Linux machine instead.
